src/transformers/models/rt_detr/modeling_rt_detr.py

```diff
@@ -0,0 +1,2502 @@
+            Logits of predicted bounding boxes coordinates in the first stage.
+    """
+
+    init_reference_points: torch.FloatTensor = None
+    last_hidden_state: torch.FloatTensor = None
+    intermediate_hidden_states: torch.FloatTensor = None
+    intermediate_reference_points: torch.FloatTensor = None
+    intermediate_logits: torch.FloatTensor = None
+    decoder_hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    decoder_attentions: Optional[Tuple[torch.FloatTensor]] = None
+    cross_attentions: Optional[Tuple[torch.FloatTensor]] = None
+    enc_topk_bboxes: Optional[torch.FloatTensor] = None
+    enc_topk_logits: Optional[torch.FloatTensor] = None
+    encoder_last_hidden_state: Optional[torch.FloatTensor] = None
+    encoder_hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    enc_outputs_class: Optional[torch.FloatTensor] = None
+    enc_outputs_coord_logits: Optional[torch.FloatTensor] = None
+    dn_meta: Optional[Dict] = None
+
+
+@dataclass
+class RTDetrObjectDetectionOutput(ModelOutput):
+    """
+    Output type of [`RTDetrForObjectDetection`].
+
+    Args:
+        loss (`torch.FloatTensor` of shape `(1,)`, *optional*, returned when `labels` are provided)):
+            Total loss as a linear combination of a negative log-likehood (cross-entropy) for class prediction and a
+            bounding box loss. The latter is defined as a linear combination of the L1 loss and the generalized
+            scale-invariant IoU loss.
+        loss_dict (`Dict`, *optional*):
+            A dictionary containing the individual losses. Useful for logging.
+        logits (`torch.FloatTensor` of shape `(batch_size, num_queries, num_classes + 1)`):
+            Classification logits (including no-object) for all queries.
+        pred_boxes (`torch.FloatTensor` of shape `(batch_size, num_queries, 4)`):
+            Normalized boxes coordinates for all queries, represented as (center_x, center_y, width, height). These
+            values are normalized in [0, 1], relative to the size of each individual image in the batch (disregarding
+            possible padding). You can use [`~RTDetrImageProcessor.post_process_object_detection`] to retrieve the
+            unnormalized (absolute) bounding boxes.
+        encoder_hidden_states (`tuple(torch.FloatTensor)`, *optional*, returned when `output_hidden_states=True` is passed or when `config.output_hidden_states=True`):
+            Tuple of `torch.FloatTensor`, being one for the output of the embeddings (logits) + one for the boxes + one
+            containing the outputs of each layer.
+    """
+
+    loss: Optional[torch.FloatTensor] = None
+    loss_dict: Optional[Dict] = None
+    logits: torch.FloatTensor = None
+    pred_boxes: torch.FloatTensor = None
+    encoder_hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+
+
+def _get_clones(module, N):
+    return nn.ModuleList([copy.deepcopy(module) for i in range(N)])
```

Comment:
Nope - absolutely not! We shouldn't be creating modules through deep copies, this is very dirty

Reply:
@amyeroberts https://github.com/huggingface/transformers/blob/f1b1379f37c6b9626bb1c795d89be4c0a606f957/src/transformers/models/deta/modeling_deta.py#L342

Well it is also have another purpose that to generate distinct module for every decoder level. This is the intended puprpose isn't it?

Reply:
Yep - and we do want to do that, but using deepcopy shouldn't be necessary - it has the behaviour of producing side-effects and is a less clear way of just instantiating new modules directly.

Reply:
I removed the deepcopy from here

Reply:
@amyeroberts Hi, amy it turns out that removing deepcopy and just making with for loop do not behave what original code expected to be. I think I need some help here! (If we solve this then it is almost done :) )

Reply:
https://discuss.pytorch.org/t/copy-deepcopy-vs-clone/55022

Reply:
Could you give some more information about the difference in behaviour between the original code and this one? 

I can see one difference between _get_clones and a for loop: the deepcopy case is copying the weights, whereas the for-loop case isn't. 

However, this shouldn't matter, because we have weight initialization triggered in the `self.post_init()` call. So, even if all the layer weights are equal when first created i.e. in the `self.class_embed = _get_clones(self.class_embed, num_pred)` call, they should all be reinitialized at the end of the execution of `__init__`

Reply:
I also exactly thought same as your word (because as you said even though we create same with for loop we initialize and load different weight for all the other method)

However, I think it has some problem with loading state_dict. Currently, I don't understand this circumstances so let me deep-dive and find out the exact reason first!